### DISCUSSION TITLE
Does the order of tags in the <head> influence SEO ranking?

### DISCUSSION THREAD

**Discussion by Cantinaband** [vercel/next.js#61033] [Help]
### Summary

We recently rebuild our website using next.js (currently version 14.0.4) and are encountering some SEO issues with the new site. It seems that Google crawlers are having difficulty detecting all the information in the <head> section of our pages.

One noticeable change with next.js is the order of elements in the <head> section. In the old app, meta-tags, structured data script-tags, and canonical were positioned at the top of the head. However, in next.js, this order is different. Some font-links, stylesheet-links, and script-tags are interspersed within the meta-tags.

The meta tags are generated using the generateMetadata function in layout.tsx | page.tsx.

We've heard that the order is crucial, and it's better for the crawler to read the meta tags first before anything else. Can anyone provide more insight into whether the order makes a difference? If so, how can we adjust the order in the head? It seems like next.js mixed up the order. 

### Additional information

Key information:
* Next.js version 14.0.4
* App Router
* Dynamic Rendering

### Example

[codesandbox](
https://codesandbox.io/p/devbox/next-head-order-z4gtwl?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrqpcnoj00083b6fngtyosvc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrqpcnoi00023b6fcmrz3bhl%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrqpcnoi00053b6fzlrbwz1i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrqpcnoi00073b6fkuqlupiq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrqpcnoi00023b6fcmrz3bhl%2522%253A%257B%2522id%2522%253A%2522clrqpcnoi00023b6fcmrz3bhl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrqpcnoi00013b6frw8e022f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrqpcnoi00013b6frw8e022f%2522%257D%252C%2522clrqpcnoi00073b6fkuqlupiq%2522%253A%257B%2522id%2522%253A%2522clrqpcnoi00073b6fkuqlupiq%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522port%2522%253A3000%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clrqpr1pb00k73b6ftsomkv8k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrqpr1pb00k73b6ftsomkv8k%2522%257D%252C%2522clrqpcnoi00053b6fzlrbwz1i%2522%253A%257B%2522id%2522%253A%2522clrqpcnoi00053b6fzlrbwz1i%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrqpcnoi00033b6fmqyecmbo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrqpcnoi00033b6fmqyecmbo%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

The screenshot show the current order of our page 

<img width="1066" alt="Screenshot 2024-01-23 at 17 28 15" src="https://github.com/vercel/next.js/assets/18029132/abe5cb69-9114-4252-a371-be852b4a08f8">

**Comment by sebastianpalmgrebban** [comment]
Facing the same issue here, in some cases we seen in Google Search Console that google isn't even bothering adding the canonical tag even if it's there. This could be an issue where the meta tags is generated to far down for Google
